rfcs/XXXX-gene-sets/text.md

```diff
@@ -0,0 +1,100 @@
+- [US2] A user should be able to upload a geneset via a csv file (see data formats below for more detail) for a dataset.
+- [US2.5] A user should be able to upload a geneset they previously downloaded from cellxgene
+- [UC3] When a user uploads a geneset, that list of genes should appear in the right side bar 
+- [US4] If the user was logged in when they uploaded a geneset that genesit should persist if they logout and back in or reload the page
+- [US5] If the gene names match those used in the dataset the user should be able to use general cellxgene functionality (color by genes, calculate differential expression levels, etc.) on those genes
+- [US6] Genesets uploaded for one dataset should be available in other datasets?
+
+### If there is a standard format for differential expression output, we should follow that. If not, the format should be as straightforward as possible while serving these above cases.
+
+### [Optional] Nonfunctional requirements
+- Users should be able to export precomputed differential expression results in real time
+- Users should be able to upload gene sets and have them immediately appear in the right side bar
+
+## Detailed Design | Architecture | Implementation
+
+
+### [Optional] Data model
+
+Fully supporting persistence of differential expression results is out of the scope of this RFC. But the data model described below has been designed to allow for differential expression persistence in the future.
+###Relational Database
+To support gene sets five tables will be added to the cellxgene relational database
+- Gene
+   - UUID (generated)
+   - GeneName
+   - OntologyId?
+- GeneGeneSetLink
+   - UUID
+   - Gene UUID
+   - GeneSet UUID
+   - Comments (optional)
+- GeneSet
+   - UUID (generated)
+   - GeneSet Name (required)
+   - Gene Count (generated)
+- GeneSetUserLink
+   - User UUID
+   - GeneSet UUID
+   - Comments (optional)
+   - Dataset UUID?
+-GeneSetDatasetLink
+   - UUID (generated)
+   - GeneSet UUID
+   - Dataset UUID
+
+Note - GeneSets will only be linked to datasets when they were uploaded with that dataset
+![Cellxgene Data Schema](imgs/Cellxgene_rds_schema.png)
+**Figure 1** Cellxgene data schema, tables to be added (as described above) are in red
+#### Imported CSV
+GeneSet CSV
+- A file containing a comma separated list of the gene set name, genes, pvalue, logfold value, 
+identifers for the cellsets that were compared to produce those values and details about why they were included. 
+Each gene should be on a new line. Only gene set (name) and gene are required fields. The name of the file will be based on the 
+user name and a timestamp rounded to the nearest second. If cells from two different datasets are being compared the datasest name 
+should be appended to the cellset label (eg dataset1.categoryA.label1)
```

Comment:
Are we adding information about the timestamp to the labels if the label is from a user-annotation?

Reply:
We could add the uuid or the uri of the user annotation that contains those categories. I am a little worried that that will just cause confusion amongst the users. It might be best to just add a warning that these values are only applicable to the category.label for the state they were generated in and if users update their annotations they should know that the diffexp values need to be recalculated. @signechambers1 what do you think?

Reply:
Agree with Madison to not include timestamps in the labels, i think the main use case for adding the timestamp to the label is to clue the user into "if the labels change your DE results might not be relevant anymore" which is better served with some sort of notification.